proposals/0000-Move-Platforms-to-Own-Packages.md

```diff
@@ -0,0 +1,273 @@
+
+The process of re-organizing these modules will, I hope, also result in more components being found that can be split out into their own repo and package as part of the slimmening.
+
+Platform-specific APIs could drop the platform suffix from their name. `PermissionsAndroid` could just be `Permissions` for example.
+
+React Native being a Yarn workspace might make it easier to make other modules that should be developed and released alongside the main project. It would be easier to make a more advanced RNTester app in-tree with its own set of dependencies, for example.
+
+The IOS-only and Android-only files and folders at the top level of the repo could get organized into their respective packages.
+
+## Detailed design
+
+### Module exports:
+
+**Components**
+
+ * `react-native`
+   * **Components**
+   * `AccessibilityInfo`
+   * `ActivityIndicator`
+   * `ART`
+   * `Button`
+   * `CheckBox` (I am unsure about this one. Android-only in main tree, but generic enough other platforms could implement it. Possible slimmening candidate?)
+   * `FlatList`
+   * `Image`
+   * `ImageBackground`
+   * `ImageEditor`
+   * `ImageStore`
+   * `KeyboardAvoidingView`
+   * `ListView` (DEPRECATED. may be removed by the time of the split)
+   * `Modal`
+   * `Picker`
+   * `SafeAreaView`
+   * `ScrollView`
+   * `SectionList`
+   * `Slider`
+   * `Switch`
+   * `RefreshControl`
+   * `StatusBar`
+   * `SwipeableFlatList`
+   * `SwipeableListView` (DEPRECATED)
+   * `Text`
+   * `TextInput`
+   * `Touchable`
+   * `TouchableHighlight`
+   * `TouchableNativeFeedback`
+   * `TouchableOpacity`
+   * `TouchableWithoutFeedback`
+   * `View`
+   * `VirtualizedList`
+   * `WebView` (DEPRECATED. [Moved to its own repo](https://github.com/react-native-community/react-native-webview))
+   * **APIs**
+   * `Alert`
+   * `Animated`
+   * `AppRegistry`
+   * `AppState`
+   * `AsyncStorage`
+   * `BackHandler`
+   * `CameraRoll`
+   * `Clipboard`
+   * `DeviceInfo`
+   * `Dimensions`
+   * `Easing`
+   * `findNodeHandle`
+   * `I18nManager`
+   * `InteractionManager`
+   * `Keyboard`
+   * `LayoutAnimation`
+   * `Linking`
+   * `NativeEventEmitter`
+   * `NetInfo`
+   * `PanResponder`
+   * `PixelRatio`
+   * `Settings`
+   * `Share`
+   * `StyleSheet`
+   * `Systrace`
+   * `TVEventHandler`
+   * `UIManager`
+   * `unstable_batchedUpdates`
+   * `Vibration`
+   * `YellowBox`
+   * **Plugins**
+   * `DeviceEventEmitter`
+   * `NativeAppEventEmitter`
+   * `NativeModules`
+   * `Platform`
+   * `processColor`
+   * `requireNativeComponent`
+   * `takeSnapshot`
+   * **Deprecated PropTypes**
+   * `ColorPropType`
+   * `EdgeInsetsPropType`
+   * `PointPropType`
+   * `ViewPropTypes`
+
+ * `react-native-ios`
+   * **Components**
+   * `DatePickerIOS`
+   * `InputAccessoryView`
```

Comment:
This is an example of something I could imagine being standard across many platforms. Also, should probably be pulled out with slimmening regardless.

Reply:
Will move this to the main package when I revise this 👍

Reply:
Honestly I think this particular one should just get moved to a community rep. But that is a conversation for slimmening. Nothing to worry about for this proposal.